packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1783,6 +1783,21 @@ export class TransactionController extends BaseControllerV1<
     const initialTx = listOfTxParams[0];
     const common = this.getCommonConfiguration(initialTx.chainId);
 
+    // We need to ensure we get the nonce using the the NonceTracker on the chain matching
+    // the txParams. In this context we only have chainId available to us, but the
+    // NonceTrackers are keyed by networkClientId. To workaround this, we attempt to find
+    // a networkClientId that matches the chainId. As a fallback, the globally selected
+    // network's NonceTracker will be used instead.
+    let networkClientId: NetworkClientId | undefined;
```

Comment:
remind me again why we shouldn't change the method signature here to include an optional networkClientId?

```
 async approveTransactionsWithSameNonce(
    listOfTxParams: (TransactionParams & { chainId: Hex, networkClientId?: NetworkClientId })[] = [],
    { hasNonce }: { hasNonce?: boolean } = {},
  ):

Reply:
previous `TransactionParams & { chainId: Hex }` just makes the existing optional chainId on TransactionParams required. chainId already being passed in from Extension. `TransactionParams & { chainId: Hex, networkClientId?: NetworkClientId }` bandages networkClientId, which doesn't already exist, onto TransactionParams. Sure, we could do this, but it seems incorrect

Reply:
Sure we don't need to tack the networkClientId into the `transactionParams` object itself but why not something like:
```
async approveTransactionsWithSameNonce(
    listOfTxParams: {
      txParams: TransactionParams & { chainId: Hex };
      networkClientId: NetworkClientId;
    }[] = [],
    { hasNonce }: { hasNonce?: boolean } = {},
  ): Promise<string | string[]> {
  ```

Reply:
I guess I just don't understand why we wouldn't pass along networkClientId information from the client as we are doing in other public entry points to this controller like `addTransaction`?